src/test/java/org/cactoos/http/HtTimedWireTest.java

```diff
@@ -76,24 +84,145 @@ public void worksFine() throws Exception {
         // @checkstyle MagicNumberCheck (1 line)
         final long timeout = 100;
-        try (Socket blocker = new Socket()) {
-            blocker.connect(this.server.getLocalSocketAddress());
-            new HtTimedWire(
-                new HtWire(
-                    this.server.getInetAddress().getHostAddress(),
-                    this.server.getLocalPort()
-                ),
-                timeout
-            ).send(new InputOf("unused"));
+        final HtTimedWire wire = new HtTimedWire(
+            input -> {
+                TimeUnit.SECONDS.sleep(timeout + 1);
+                return input;
+            },
+            timeout
+        );
+        new Assertion<>(
+            HtTimedWireTest.TIMEOUT,
+            () -> wire.send(new InputOf(HtTimedWireTest.UNUSED)),
+            new TimeoutExceptionMatcher()
+        )
+            .affirm();
+    }
+
+    // @checkstyle MagicNumberCheck (1 line)
+    @Test(timeout = 1000)
+    public void failsAfterTimeoutCheckIfWait() throws Exception {
+        final long timeout = 100;
+        final long sleep = 10 * timeout;
+        final HtTimedWire wire = new HtTimedWire(
+            input -> {
+                TimeUnit.SECONDS.sleep(sleep);
+                return input;
+            },
+            timeout
+        );
+        final long current = System.currentTimeMillis();
+        new Assertion<>(
+            HtTimedWireTest.TIMEOUT, () -> wire.send(
+            new InputOf(HtTimedWireTest.UNUSED)
+        ), new AllOf(
+            new TimeoutExceptionMatcher(),
+            new TypeSafeDiagnosingMatcher<Scalar<Input>>() {
+                @Override
+                protected boolean matchesSafely(
+                    final Scalar<Input> item,
+                    final Description description) {
+                    final long failed = System.currentTimeMillis();
+                    return failed - current < sleep;
+                }
+
+                @Override
+                @SuppressWarnings("PMD.UncommentedEmptyMethodBody")
+                public void describeTo(final Description description) {
+                }
+            }));
+    }
+
+    /**
+     * Class that matches that {@link java.util.concurrent.TimeoutException}
+     * is thrown.
+     * @todo #87:60m For now we can't use
+     *  {@link org.llorllale.cactoos.matchers.Throws}
+     *  to match exception message because message in the exception is null.
+     *  I've tried to update to the new version of matchers and
+     *  use constructor of Throws with {@link org.hamcrest.Matcher},
+     *  but it leads update the cactoos
+     *  and after that I can't build a project.
+     *  So needs to update dependencies, fix conflicts and replace
+     *  this class with {@link org.llorllale.cactoos.matchers.Throws}.
+     *  And needs try to remove {@link java.lang.SuppressWarnings} in
+     *  {@link org.cactoos.http.HtTimedWireTest}.
+     */
+    private final class TimeoutExceptionMatcher extends
```

Comment:
@aivinog1 leave a `@todo` to replace this with llorllale/cactoos-matchers#133

Reply:
Are you sure? This matcher only matches that exception will throw not about timeouts.

Reply:
@aivinog1 oh I see

Reply:
@aivinog1 this means you can use `Throws` from cactoos-matchers

Reply:
@aivinog1 please rename to `ThrowsTimeoutException`

Reply:
@llorllale Done.